lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +47,48 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if not any(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+
+        list_of_ids = []
+        for file in self.path_to_raw_txt_data.glob("*_raw.txt"):
+            print(file)
+            if not file.stat().st_size:
+                raise InconsistentDatasetError
+
+            list_of_ids.append(int(file.name[:file.name.index('_')]))
+            if sorted(list_of_ids) != list(range(1, len(list_of_ids) + 1)):
```

Comment:
Is it necessary to check the order after every single iteration? Could it be done all at once after all the IDs are collected?